sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/CheckpointManager.cs

```diff
@@ -55,7 +55,7 @@ public class CheckpointManager
         ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
-        public Task UpdateCheckpointAsync(EventData eventData) => UpdateCheckpointAsync(eventData.Offset, eventData.SequenceNumber);
+        public Task UpdateCheckpointAsync(EventData eventData) => UpdateCheckpointAsync(eventData.Offset.Value, eventData.SequenceNumber.Value);
```

Comment:
What if `eventData.SystemProperties` is null? The user shouldn't be doing this, but if they create an `EventData` from the public constructor and pass it to `UpdateCheckpointAsync`, it will throw an exception.

It will be thrown from one of the user's methods (`Initialize`, `Close`, `ProcessEvents` or `ProcessError`). Maybe we should just treat it as a user exception as well and handle it normally in the Event Processor.

No action needed. I'm just adding some relevant information.

Reply:
You have the same problem today.   If the property doesn't exist in the key/value pairs of the system properties, an exception is thrown.    You may want to guard against that by making the parameters nullable and then doing a null check.

Reply:
Nice. I'll take a note.